src/intra_edge.rs

```diff
@@ -103,58 +158,57 @@ impl EdgeBranch {
     const fn new(edge_flags: EdgeFlags, bl: BlockLevel) -> Self {
         let o = edge_flags;
         let h = [
-            edge_flags | EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_LEFT_HAS_BOTTOM,
+            edge_flags.union(EdgeFlags::LEFT_HAS_BOTTOM),
+            edge_flags.intersection(EdgeFlags::LEFT_HAS_BOTTOM),
         ];
         let v = [
-            edge_flags | EDGE_TOP_HAS_RIGHT,
-            edge_flags & EDGE_TOP_HAS_RIGHT,
+            edge_flags.union(EdgeFlags::TOP_HAS_RIGHT),
+            edge_flags.intersection(EdgeFlags::TOP_HAS_RIGHT),
         ];
         let node = EdgeNode { o, h, v };
 
         let h4 = [
-            edge_flags | EDGE_LEFT_HAS_BOTTOM,
-            EDGE_LEFT_HAS_BOTTOM
-                | (if bl == BL_16X16 {
-                    edge_flags & EDGE_I420_TOP_HAS_RIGHT
-                } else {
-                    0 as EdgeFlags
-                }),
-            EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_LEFT_HAS_BOTTOM,
+            edge_flags.union(EdgeFlags::LEFT_HAS_BOTTOM),
+            EdgeFlags::LEFT_HAS_BOTTOM.union(
+                (edge_flags.intersection(EdgeFlags::I420_TOP_HAS_RIGHT)).select(bl == BL_16X16),
+            ),
+            EdgeFlags::LEFT_HAS_BOTTOM,
+            edge_flags.intersection(EdgeFlags::LEFT_HAS_BOTTOM),
         ];
 
         let v4 = [
-            edge_flags | EDGE_TOP_HAS_RIGHT,
-            EDGE_TOP_HAS_RIGHT
-                | (if bl == BL_16X16 {
-                    edge_flags & (EDGE_I420_LEFT_HAS_BOTTOM | EDGE_I422_LEFT_HAS_BOTTOM)
-                } else {
-                    0 as EdgeFlags
-                }),
-            EDGE_TOP_HAS_RIGHT,
-            edge_flags & EDGE_TOP_HAS_RIGHT,
+            edge_flags.union(EdgeFlags::TOP_HAS_RIGHT),
+            EdgeFlags::TOP_HAS_RIGHT.union(
+                edge_flags
+                    .intersection(EdgeFlags::union_all([
+                        edge_flags.intersection(EdgeFlags::I420_LEFT_HAS_BOTTOM),
+                        edge_flags.intersection(EdgeFlags::I422_LEFT_HAS_BOTTOM),
+                    ]))
+                    .select(bl == BL_16X16),
+            ),
+            EdgeFlags::TOP_HAS_RIGHT,
+            edge_flags.intersection(EdgeFlags::TOP_HAS_RIGHT),
         ];
 
         let tls = [
-            EDGE_TOP_HAS_RIGHT | EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_LEFT_HAS_BOTTOM,
-            edge_flags & EDGE_TOP_HAS_RIGHT,
+            EdgeFlags::TOP_HAS_RIGHT.union(EdgeFlags::LEFT_HAS_BOTTOM),
```

Comment:
`union_all`? I think it looks best with `.union` when the LHS is `edge_flags`, but when the LHS and RHS are equivalent, a symmetric way seems nicer.

Reply:
yes totally agree!

Reply:
though actually this could be `EdgeFlags::all()`

Reply:
> though actually this could be `EdgeFlags::all()`

Oh true, even better.